accounting/invoice/tests/test_choices.py

```diff
@@ -35,3 +43,97 @@ def test_approved_choices(self):
+        self.assertEqual(choices.InvoiceNumberingScheme.default_opencraft_year_month(), 'OC-2018-01')
+
+    def test_default_opencraft_number(self):
+        """ The default value of the `opencraft_number` choice corresponds to the correct prefix and number combo. """
+        self.assertEqual(choices.InvoiceNumberingScheme.default_opencraft_number(), 'OC-1')
+
+    @ddt.data(
+        ('2018-01', '2018-02'),
+        ('2018-12', '2019-01'),
+    )
+    @ddt.unpack
+    def test_increment_default(self, value, incremented_value):
+        """ Incrementing the `default` choice gives the correct year and month, including for wrap-arounds. """
+        self.assertEqual(choices.InvoiceNumberingScheme.increment_default(value), incremented_value)
+
+    @ddt.data(
+        ('3', '4'),
+        ('000003', '4'),
+    )
+    @ddt.unpack
+    def test_increment_number(self, value, incremented_value):
+        """ Incrementing the `number` choice gives the correct number, including with leading 0s. """
+        self.assertEqual(choices.InvoiceNumberingScheme.increment_number(value), incremented_value)
+
+    @ddt.data(
+        ('2018-01-4', '2018-02-5'),
+        ('2018-12-10', '2019-01-11'),
+        ('2018-12-0001', '2019-01-2'),
+    )
+    @ddt.unpack
+    def test_increment_year_month_number(self, value, incremented_value):
+        """ Incrementing the `year_month_number` choice gives the correct year, month, and number combo,
+         including for wrap-arounds and with leading 0s. """
+        self.assertEqual(choices.InvoiceNumberingScheme.increment_year_month_number(value), incremented_value)
+
+    @ddt.data(
+        ('OC-2018-01', 'OC-2018-02'),
+        ('OC-2018-12', 'OC-2019-01'),
+    )
+    @ddt.unpack
+    def test_increment_opencraft_year_month(self, value, incremented_value):
+        """ Incrementing the `opencraft_year_month` choice gives the correct prefix and year, month combo,
+         including for wrap-arounds. """
+        self.assertEqual(choices.InvoiceNumberingScheme.increment_opencraft_year_month(value), incremented_value)
+
+    @ddt.data(
+        ('OC-3', 'OC-4'),
+        ('OC-000003', 'OC-4'),
+    )
+    @ddt.unpack
+    def test_increment_opencraft_number(self, value, incremented_value):
+        """ Incrementing the `opencraft_number` choice gives the correct prefix and number including for leading 0s. """
+        self.assertEqual(choices.InvoiceNumberingScheme.increment_opencraft_number(value), incremented_value)
+
+    @ddt.data(
+        (choices.InvoiceNumberingScheme.default, None, '2018-02'),
+        (choices.InvoiceNumberingScheme.default, '2018-12', '2019-01'),
+        (choices.InvoiceNumberingScheme.number, '2', '3'),
+        (choices.InvoiceNumberingScheme.year_month_number, '2018-12-3', '2019-01-4'),
+        (choices.InvoiceNumberingScheme.opencraft_year_month, 'OC-2018-09', 'OC-2018-10'),
+        (choices.InvoiceNumberingScheme.opencraft_number, 'OC-0002', 'OC-3'),
```

Comment:
@UmanShahzad If someone wants a certain number of leading zeros (or a certain number of digits in the number part of this format), shouldn't we preserve that?

Reply:
@itsjeyd I would want to but this adds complexity to incrementing the values. I think only @tomaszgy has it this way -- would this be OK with you, @tomaszgy ?

Reply:
@UmanShahzad Please excuse but I'm not sure what I'm being asked here :) Could you please say a bit more?

Reply:
@tomaszgy The question is whether you'd be OK with moving from a numbering scheme that pads numbers with zeroes to a scheme that doesn't? For example, if your last invoice for OpenCraft was `OC-0004`, would you be OK with future invoices being numbered `OC-5`, `OC-6`, etc.?

For context: This PR implements some logic for auto-incrementing invoice numbers according to different schemes. The current version of the code would move you from the scheme involving zeroes to the scheme that doesn't. It would require additional effort to implement an approach that preserves the leading zeroes.

Let me know if that makes sense :)

Reply:
@tomaszgy On your invoices you use OC-0001 as the invoice number. For the next invoice, would you be OK using OC-2, instead of OC-0002? (Trying to use the latter makes things more difficult for the automation code).

Reply:
@tomaszgy Sorry, I meant yyyy-mm-{number}, not OC-{number} :) 

But yep, we're going to start using it now.

Reply:
@UmanShahzad, cool, let'd go ahead with the new schema then :)